File: test_file.py
def hello_world():
    print("Hello, world!")
    return "greeting"

class Calculator:
    def __init__(self):
        self.value = 0
        self.history = []

    def add(self, number):
        """Add a number to the current value"""
        self.value += number
        self.history.append(f"Added {number}")
        return self.value

    def multiply(self, factor):
        """Multiply current value by factor"""
        old_value = self.value
        self.value *= factor
        self.history.append(f"Multiplied by {factor}")
        return self.value

def main():
    calc = Calculator()
    result = calc.add(5)
    print(f"Result: {result}")
    
    result = calc.multiply(3)
    print(f"Final result: {result}")

if __name__ == "__main__":
    main()

File: test_file.rs
fn main() {
    let result = calculate_sum(5, 10);
    println!("The result is: {}", result);
}

fn calculate_sum(a: i32, b: i32) -> i32 {
    let sum = a + b;
    println!("Adding {} and {}", a, b);
    sum
}

struct Calculator {
    value: i32,
    history: Vec<String>,
}

impl Calculator {
    fn new() -> Self {
        Calculator {
            value: 0,
            history: Vec::new(),
        }
    }

    fn add(&mut self, number: i32) -> i32 {
        self.value += number;
        self.history.push(format!("Added {}", number));
        self.value
    }

    fn get_result(&self) -> i32 {
        println!("Current result: {}", self.value);
        self.value
    }
}